src/strands/tools/mcp/mcp_from_config.py

```diff
@@ -0,0 +1,228 @@
+"""MCP server configuration loading utilities."""
+
+import json
+import re
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from pathlib import Path
+from typing import Callable, Dict, List, Optional
+from urllib.parse import urlparse
+
+from mcp import StdioServerParameters
+from mcp.client.sse import sse_client
+from mcp.client.stdio import stdio_client
+from mcp.client.streamable_http import streamablehttp_client
+
+from .mcp_client import MCPClient
+from .mcp_types import MCPTransport
+
+
+class MCPTransportType(Enum):
+    """MCP transport types."""
+
+    STDIO = "stdio"
+    STREAMABLE_HTTP = "streamable-http"
+    SSE = "sse"
+
+
+@dataclass
+class MCPTransportConfig(ABC):
+    """Base configuration for MCP Transports."""
+
+    name: str
+    timeout: float = 60000
+    transport_type: Optional[MCPTransportType] = None
```

Comment:
I don't think this is needed - you should be able to check each derived type instead.

E.g. `isinstance(config, StdioTransportConfig)`